src/TwoOutOfThreeColumnVisitor.cpp

```diff
@@ -0,0 +1,148 @@
+                        regions.push_back(ioGrid.getRegion(3*((i+0)%3) + main_col));
+                        regions.push_back(ioGrid.getRegion(3*((i+1)%3) + main_col));
+                        regions.push_back(ioGrid.getRegion(3*((i+2)%3) + main_col));
+                        
+                        for(int value = 1; value<=9; ++value) {
+                                // test for all different values
+                                //
+                                if(regions[0].isValuePresent(value)){continue;}
+
+                                // value is not present in the ith region
+                                // find the col where it should be
+                                
+                                possibleColumns = vector<bool>(3, false);
+                                position = regions[1].valuePosition(value);
+                                if (position.second != -1) {
+                                        possibleColumns[position.second] = true;
+                                }
+                                position = regions[2].valuePosition(value);
+                                if (position.second != -1) {
+                                        possibleColumns[position.second] = true;
+                                }
+
+
+
+                                // find the rows where it should be
+                                possibles = std::vector<std::vector<bool>>();
+                                for(int j=0; j<3; ++j) {
+
+                                        // if the columns already have this value from another Region, pass
+                                        if(possibleColumns[j]) {
+                                                possibles.push_back(vector<bool>(3, true));
+                                                continue;
+                                        } else {
+                                                possibles.push_back(vector<bool>(3, false));
+                                        }
+                                        
+                                        // This is the same for all j
+                                        possibles[j][0] = ioGrid.getRegion(3*((i+0)%3) + (main_col+1)%3).topRow().isValuePresent(value) || ioGrid.getRegion(3*((i+0)%3) + (main_col+2)%3).topRow().isValuePresent(value); 
+                                        possibles[j][1] = ioGrid.getRegion(3*((i+0)%3) + (main_col+1)%3).middleRow().isValuePresent(value) || ioGrid.getRegion(3*((i+0)%3) + (main_col+2)%3).middleRow().isValuePresent(value); 
+                                        possibles[j][2] = ioGrid.getRegion(3*((i+0)%3) + (main_col+1)%3).bottomRow().isValuePresent(value) || ioGrid.getRegion(3*((i+0)%3) + (main_col+2)%3).bottomRow().isValuePresent(value); 
+                                        
+                                        // Cannot be place in cells that are not empty
+                                        if(j == 0) {
```

Comment:
use this instead: https://github.com/albanD/sudoku-solver/blob/master/src/RegionHolder.cpp#L54
That way we can keep the triple holder const